src/config/session.c

```diff
@@ -80,6 +83,74 @@ read_environment_file(const char *filename)
 	return true;
 }
 
+static char *
+valid_env_path(const char *prefix, const char *name)
```

Comment:
Consider splitting this into `has_valid_file_extension()` and `path_join()`

Reply:
I moved the extension check to a more general string helper called `str_endswith`. This also behaves with NULL or empty arguments, so we can eliminate some up-front checks in this function.